client/network/src/config.rs

```diff
@@ -255,58 +255,76 @@ pub fn parse_addr(mut addr: Multiaddr) -> Result<(PeerId, Multiaddr), ParseErr>
 	Ok((who, addr))
 }
 
-/// Splits a RpcAddress into a SocketAddr and PeerId.
-pub fn parse_str_rpc_addr(rpc_addr_str: &str) -> Result<(PeerId, std::net::SocketAddr), ParseErr> {
+/// Splits a string address into a SocketAddr, PeerId and QkdAddress.
+pub fn parse_str_gen_addr(rpc_addr_str: &str) -> Result<(PeerId, std::net::SocketAddr, url::Url), ParseErr> {
 	use regex::Regex;
-	
-	let rpc_addr_regex = Regex::new(r"^/([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{1,5}/[a-zA-z0-9]{40,}$").unwrap();
-    if rpc_addr_regex.is_match(rpc_addr_str) {
+
+	let gen_addr_regex = Regex::new(r"^https?://([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{1,5}(/[a-zA-Z0-9]+){0,5}/[a-zA-z0-9]{40,}$").unwrap();
```

Comment:
I see a few problems here:
- it can't be https (at least for now), because of the problem with certificates
- QKD config knows about RPC and vice versa, but they shouldn't.
- we should rethrow the error instead of .unwrap() on regex
- naming should be improved: rpc_addr_split (why rpc since it contains also qkd path? while names like split, vec, url say nothing about what they contain, try to e more explicit)

I suggest to do this struct this way:
```
pub struct AddrWithPeerId {
	pub host: std::net::SocketAddr,
    pub path: Option<String>,
	pub peer_id: PeerId,
}
```
Then we for rpc config will have path as `None` and qkd config will have path as `Some("path/to/alice")`